### ISSUE TITLE
adding music while background is lightened up in java script 

### ISSUE THREAD

**Issue by richardmsk** [richardmsk/click#3]
and also making the clicking sound effects overlay each other when clicked on rapidly, and also the problem with locking the next page button in place and size.

**Comment by sehsarah** [comment]
> adding music while background is lightened up

You can use the same method you're using for the click sound

`document.getElementById('audio_play').play();` https://github.com/richardmsk/click/blob/main/index.html#L25
which works with an the audio element that has that `id` (make sure you use a different `id` than the click sound too)

Then instead of including it in your HTML via `onClick` you instead need to add it within your JS script, inside the click handler function as a line after the class toggle https://github.com/richardmsk/click/blob/main/index.html#L36

However you'll need logic added that says "if playing stop, and if not playing, play" because there's not an equivalent `toggle` method for audio

> making the clicking sound effects overlay each other when clicked on rapidly

Not sure about this one off the top of my head

> problem with locking the next page button in place and size.

Can you include a design / more information about what you're trying to achieve and what the issues are (e.g. use annotation on the design to explain this)

@richardmsk

**Comment by richardmsk** [comment]
I don't quite get it, where do I put the onclick in? 

lightswitch.addEventListener('click', function(e){
        document.body.classList.toggle('cl-switch--on');
        onClick="document.getElementById('audio_play').play(); return false;"/

this doesn't work, and do I also put the audio src codes under js or just under HTML?

**Comment by richardmsk** [comment]
@sehsarah

**Comment by sehsarah** [comment]
1. audio element goes in HTML
2. edited your snippet below 

```
lightswitch.addEventListener('click', function(e){
        document.body.classList.toggle('cl-switch--on');
        document.getElementById('audio_play').play()
});
```

but since you're already using an audio element with id `audio_play` 
you will need to change that id and create a new audio element with a new id

**Comment by richardmsk** [comment]
After adding said code, the switch no longer changes, and music also don't play, only the audio queue for clicking is working 

  (function() {
    var lightswitch = document.getElementById('cl-switch');

    lightswitch.addEventListener('click', function(e){
        document.body.classList.toggle('cl-switch--on');
        document.getElementById('music').play()
});
});

**Comment by richardmsk** [comment]
@sehsarah 
if not possible I will just give up the idea.

**Comment by sehsarah** [comment]
Have u made sure to add an audio element with the id `music`

If you open console and screenshot that here, it should display the error you're encountering

**Comment by richardmsk** [comment]
<html>
<head>
  <meta charset="utf-8">
  <meta http-equiv="Cache-Control" content="no-store">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome1">
  <meta name="viewport" content="width=deivce-width, initial-scale=1.0">
  <title>Click</title>
  <link rel="stylesheet" href="site.css">
  <link rel="shortcut icon" type="image/png" href="favicon.ico">
  <style>
  @font-face {
    font-family: PIXEAL;
    src: url( Minecraft.ttf);
  }
</style>
</head>
<body>
  <audio id="audio_play" style="display:none">
  <source src="switch.mp3" type="audio/mpeg"> </audio>
  <audio id="music" style="display:none">
  <source src="music1.mp3" type="audio/mpeg"> </audio>
  <div class="text-container">
    <a href="sub-01.html"><p style="color:#ff9999">NEXT</p></a>
  </div>
  <div id="cl-switch" class="cl-switch-container">
    <div class="cl-switch-on">
      <img src="assets/switch-on-croped.png " onClick="document.getElementById('audio_play').play(); return false;" />
    </div>
    <div class="cl-switch-off">
      <img src="assets/switch-off-croped.gif" onClick="document.getElementById('audio_play').play(); return false;" />
    </div>
  </div>
  <script>
  (function() {
    var lightswitch = document.getElementById('cl-switch');

    lightswitch.addEventListener('click', function(e){
        document.body.classList.toggle('cl-switch--on');
        document.getElementById('music').play()
});
});
  </script>
</body>
</html>

**Comment by richardmsk** [comment]
heres the src files my music file is named music1.mp3
<audio id="audio_play" style="display:none">
  <source src="switch.mp3" type="audio/mpeg"> </audio>
  <audio id="music" style="display:none">
  <source src="music1.mp3" type="audio/mpeg"> </audio>

**Comment by richardmsk** [comment]
the problem is not the music is playing as of yet, its the image switching being bugged after the update

![image](https://user-images.githubusercontent.com/91919871/138026486-cd52ed31-d747-459b-bfa6-b1068588b929.png)

**Comment by richardmsk** [comment]
@sehsarah

**Comment by sehsarah** [comment]
can u screenshot the console error you'll be getting when u load the site? @richardmsk

**Comment by richardmsk** [comment]
there are no errors
![image](https://user-images.githubusercontent.com/91919871/138026604-3cbb3554-f412-40a1-acd0-adc265093d30.png)

**Comment by richardmsk** [comment]
the yellow one is in <head> that don't matter

**Comment by richardmsk** [comment]
![image](https://user-images.githubusercontent.com/91919871/138026667-c6f465d2-91c6-4755-9327-d41def182611.png)

**Comment by sehsarah** [comment]
is the class from line 38 in your screenshot being added to the body?

**Comment by richardmsk** [comment]
no, its under js function

**Comment by richardmsk** [comment]
![image](https://user-images.githubusercontent.com/91919871/138026861-d6771d22-580d-460d-b98e-e5dcc38da96b.png)

**Comment by richardmsk** [comment]
could it be the file size of the mp3? its 4mbs

**Comment by sehsarah** [comment]
I'm asking if `cl-switch--on` is getting added to the body when you run your site seeing the issue you're describing

You can use inspect element to check this

**Comment by sehsarah** [comment]
And yes that's a large audio file @richardmsk

**Comment by sehsarah** [comment]
Probably try and cut a short loop and create a smaller file. Are your other sound effect files that large too?

**Comment by richardmsk** [comment]
no it is not added in the HTML body

**Comment by richardmsk** [comment]
yeah I can get samller ones

**Comment by richardmsk** [comment]
yeah still broken, must be something with the js codes, I deleted the js addons and it went back to normal, I think I could be becoz its running 2 instances at once and somehow bugging the system.

**Comment by richardmsk** [comment]
I will just leave it as that for now, if you have free time maybe we can do a call, else I will just leave it. Thanks for the time anyways.

**Comment by sehsarah** [comment]
Maybe just post your html in here so i can have a quick look and see if i can tell what's happening @richardmsk

**Comment by richardmsk** [comment]
sent though teams, this section don't allow for file uploads and somehow copying it in to comment makes it disappear
